plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -464,8 +525,17 @@ public function get_download_permissions_granted( $order ) {
 	 * @param bool      $save Whether to persist changes to db immediately or not.
 	 */
 	public function set_download_permissions_granted( $order, $set, $save = true ) {
```

Comment:
(Another discussion that applies to all props)

When we initially added the `$save` arg it was for performance reasons, as we didn't want the datastore level methods to always trigger a save, but now that these props can be set at the order level, wouldn't it be more "compatible" if the COT datastore methods match those in CPT exactly (i.e. get rid of `$save`) and always call `$order->save()`?

We could mark these methods as deprecated in COT and suggest the order-level methods are used instead, but if you do call the datastore-level ones you'll get the exact same behavior as CPT, which seems reasonable to me.
Also, it'd avoid the "no op" scenario you've covered with a `_doing_it_wrong()`.

What do you think?

Reply:
Makes sense, not that we are storing data in order prop, they can be used when performance is needed. Removed the param in https://github.com/woocommerce/woocommerce/pull/34627/commits/0cb6cadf0a6054724e02074c552ca1c66bffc6d2